### PR TITLE
One Makefile to rule them all.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: cerealize/KnowSheetDir.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+#
+# Goes up the directory tree until "KnowSheet/" sub-directory is found.
+# Prints its path.
+
+dir=$(cd $(dirname $0) && pwd)
+subdir=KnowSheet
+
+while [ ! -d ${dir}/${subdir} ] ; do
+  [ "$dir" == "/" ] && dir="" && break
```
Comment (kt97679) [inline cerealize/KnowSheetDir.sh]: dir="" not necessary here since you check if directory exists after loop

Comment (dkorolev) [inline cerealize/KnowSheetDir.sh]: Thanks, fixed.

**Comment by dkorolev** [comment]
@deathbaba PTAL.

**Comment by deathbaba** [comment]
Why do you need all these dir print scripts? Can't you do it without copy-paste?

**Comment by dkorolev** [comment]
@deathbaba Please suggest a solution without copy-paste. I'm all for it, but git/GitHub do not support symlinks, and modern build systems are far from perfect.

**Comment by deathbaba** [comment]
git/github support symlinks.

**Comment by dkorolev** [comment]
Just checked, indeed, thanks! Hang on for a few mins for an update to this pull request ...

**Comment by dkorolev** [comment]
@deathbaba : Please take another look.

**Pull request merged**